src/common/config.cc

```diff
@@ -276,16 +323,20 @@ int md_config_t::parse_config_files_impl(const std::list<std::string> &conf_file
     }
   }
 
+  std::string error_message;
   std::vector <std::string> my_sections;
   _get_my_sections(my_sections);
   for (auto& opt: *config_options) {
     std::string val;
     int ret = _get_val_from_conf_file(my_sections, opt.name, val, false);
     if (ret == 0) {
-      set_val_impl(val.c_str(), &opt);
+      set_val_impl(val, &opt, &error_message);
+      if (warnings != nullptr && !error_message.empty()) {
+        *warnings << error_message << std::endl;
```

Comment:
I would be nice to prefix the error message with something like 'parse config file: opt.name: '. Otherwise it looks a little confusing where the errors comes from:
```
% ./bin/rbd info test5
unknown rbd feature SSS
unknown rbd feature SSS
unknown rbd feature SSS
```